### PR TITLE
Fix major bug with tornado

### PR DESCRIPTION
**Pull request by Rafiot**
I misunderstood the way handle_stream works: when you do `self.stream = stream`,
`self.stream` will be overwritten everytime a new client connects, which
means you endup with the second client receiving the response from the
query made by the first.

I also added a proper timeout in case the client of uwhoisd never sends
a new line and the stream stays open forever.

Ressources: http://alexapps.net/python-tornado-simple-tcp-server and https://github.com/tornadoweb/tornado/issues/689

### CODE REVIEW

**Comment by kgaughan** [comment]
Been busy over the last few days, but I'll take a look over the patch this evening to make sure I understand it.

**Comment by Rafiot** [comment]
Sure, I'll add some in-line comments

File: uwhoisd/net.py [hunk 1]
```diff
@@ -76,37 +78,104 @@ def whois(self, query):
         return to_return
 
 
+@contextlib.contextmanager
+def auto_timeout(self, timeout):
```
Comment (Rafiot) [inline uwhoisd/net.py]: Asynchronously set a timeout and close the stream if the timeout is reached.

File: uwhoisd/net.py [hunk 2]
```diff
@@ -76,37 +78,104 @@ def whois(self, query):
...
-            whois_query = yield self.stream.read_until_regex(b'\s')
-            whois_query = whois_query.decode().strip().lower()
-            if not utils.is_well_formed_fqdn(whois_query):
-                whois_entry = "; Bad request: '{0}'\r\n".format(whois_query)
-            else:
-                whois_entry = self.whois(whois_query)
-            yield self.stream.write(whois_entry.encode())
-        except tornado.iostream.StreamClosedError:
-            logger.warning('Connection closed by %s.', address)
+            connection = ClientHandler(stream, self.whois, address,
```
Comment (Rafiot) [inline uwhoisd/net.py]: Immediately spawn the stream in a thread and move on.

**Comment by Rafiot** [comment]
FYI, this patch is now running on my production instance for a week, didn't fail and processed many thousands of queries.

**Comment by kgaughan** [comment]
@Rafiot Could you rebase off of the current master branch? Due to other commitments over the years, I haven't been able to do any development on uwhoisd up until now, but it'd be good to get this merged.

**Comment by kgaughan** [comment]
Also, be sure to run the code through black.

**Comment by Rafiot** [comment]
I'll try to get that sorted. I diverged quite a bit from your code base, so can you please confirm you're still working on it, and will be in the next weeks? I want to avoid working on that and having it stalled for years.

**Pull request closed without merging**